### DISCUSSION TITLE
Networking options

### DISCUSSION THREAD

**Discussion by raghavendra-talur** [lima-vm/lima#2388] [General]
I am exploring the networking options in Lima.  Here is what I have found so far:
- user-v2 supports VM to VM network but no access from the host
- shared with socket_vmnet supports VM to VM network as well as host access but needs external components and sometimes has issues with split DNS implemented by firewalls 
- vzNAT supports only access from the host. No VM to VM communication 

Is there anyway to get host access in user-v2 or allow for VM to VM communication in vzNAT?

My goal is to create a 3 k8s environment that is also accessible from the host. I was able to get very close to it by having both vzNAT and user-v2 interfaces in the VM and configure kubernetes API server to listen on both but the services themselves only use node port on the vzNAT interface and therefore aren't accessible from the host.

Any recommendations on how to solve this problem?

**Comment by louhisuo** [comment]
Take a look to [k8s-on-macos](https://github.com/louhisuo/k8s-on-macos) to evaluate if it works for you as it is based on shared network.
